### ISSUE TITLE
Option to default to platform brightness on Android Q

### ISSUE THREAD

**Issue by pbaker5** [Norbert515/dynamic_theme#21]
On Android Q I'd like to use the OS's Theme option (which is exposed via MediaQuery.of(context).platformBrightness) instead of having another checkbox in my settings section of my app. Can we please have an option to use that as the defaultBrightness? Then I can simply check for Android Q and hide the dark mode checkbox in my settings screen.

**Comment by Norbert515** [comment]
Do you think forcing the app to be dark when the platformBrightness is set to dark is okay (without allowing the user to switch back in the app)?

**Comment by pbaker5** [comment]
Yes. I think the OS setting makes the app setting redundant.

**Comment by Norbert515** [comment]
We'd still want to have the ability to overwrite that (especially for backward compatibility sake!).

I'm pretty busy over the next couple of days, if you want to, you can open a PR for this, if not I'll look into it once I find the time.

**Comment by sterrenburg** [comment]
> Yes. I think the OS setting makes the app setting redundant.

I agree, perhaps adding a UI toggle for overriding the OS setting would be nice too.

**Comment by TheLastGimbus** [comment]
This should be easy with flutter. Just set `ThemeMode.system`. Or it's not, I don't fully know how dynamic_theme works under the hood.

**Comment by sterrenburg** [comment]
The easiest way seems to be indeed setting [`MaterialApp.themeMode`](https://api.flutter.dev/flutter/material/MaterialApp/themeMode.html).

This can be quickly implemented with Provider. This [untested example](https://github.com/sterrenburg/flutterhole/blob/a561d353f74beecc91d26273cc208ce6e446639a/lib/features/settings/presentation/notifiers/theme_mode_notifier.dart#L6-L15) seems to work for me.
